cast-receiver/src/status_ui_handler.ts

```diff
@@ -1,104 +1,102 @@
+  private static readonly IDLE_STATUS_ICON = "cast";
+  private static readonly CASTING_STATUS_ICON = "cast";
+  private static readonly LOADER_STATUS_ICON = "loader";
+
+  private static readonly IDLE_STATUS_MSG = "Ready to cast";
+  private static readonly CASTING_STATUS_MSG = "A device is casting";
 
   private container: HTMLDivElement;
-  private readyToCastElement: HTMLSpanElement;
+  private statusElements: Map<string, HTMLElement> = new Map();
 
   constructor(statusContainer: HTMLDivElement) {
     this.container = statusContainer;
-    this.readyToCastElement = this.createReadyToCastElement();
+    this.statusElements.set(
+      StatusUIHandler.IDLE_STATUS_ID,
+      this.createStatusElement(
+        StatusUIHandler.IDLE_STATUS_ID,
+        StatusUIHandler.IDLE_STATUS_ICON,
+        StatusUIHandler.IDLE_STATUS_MSG
+      )
+    );
+
+    this.statusElements.set(
+      StatusUIHandler.CASTING_STATUS_ID,
+      this.createStatusElement(
+        StatusUIHandler.CASTING_STATUS_ID,
+        StatusUIHandler.CASTING_STATUS_ICON,
+        StatusUIHandler.CASTING_STATUS_MSG
+      )
+    );
+
+    this.statusElements.set(
+      StatusUIHandler.LOADER_STATUS_ID,
+      this.createStatusElement(
+        StatusUIHandler.LOADER_STATUS_ID,
+        StatusUIHandler.LOADER_STATUS_ICON,
+        ""
+      )
+    );
   }
 
   private createIcon(id: string): HTMLElement {
     const template = document.createElement("template");
-    template.innerHTML = `<svg id="${id}" class="sound"><use xlink:href="${Icons[id]}" /><svg>`;
+    template.innerHTML = `<svg id="${id}" class="icon"><use xlink:href="${Icons[id]}" /><svg>`;
     return template.content.firstChild as HTMLElement;
   }
 
-  private createReadyToCastElement(): HTMLSpanElement {
-    const element = document.createElement("span");
-    element.classList.add("caption");
-    element.id = StatusUIHandler.READY_TO_CAST_ID;
-    element.appendChild(this.createIcon("cast"));
-    element.appendChild(
-      document.createTextNode(StatusUIHandler.READY_TO_CAST_MSG)
-    );
-
+  private createStatusElement(
+    id: string,
+    iconID: string,
+    message: string
+  ): HTMLSpanElement {
+    const element = document.createElement("div");
+    element.classList.add("statusline", "caption");
+    element.id = id;
+    element.appendChild(this.createIcon(iconID));
+    element.appendChild(document.createTextNode(message));
     return element;
   }
 
   /**
-   * Puts the ready to cast message on status display. If status display is already
-   * showing ready to cast message, the method does nothing.
+   * Enables the status with given ID.
+   *
+   * @param statusID ID of status to toggle. must be one of
+   * StatusUIHandler.IDLE_STATUS_ID or StatusUIHandler.CASTING_STATUS_ID
+   * @param enabled whether to display or hide the given status
    */
-  showIdleStatus(): void {
-    if (this.isShowingIdleStatus() === true) {
+  enableStatus(statusID: string, enabled: boolean): void {
```

Comment:
Codacy found an issue: [Method 'enableStatus' has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/manual/ashutoshgngwr/noice/pullRequest?prid=6042010)